adyenv6b2ccheckoutaddon/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -534,16 +540,47 @@ public void handlePaymentForm(AdyenPaymentForm adyenPaymentForm, BindingResult b
             return;
         }
 
+        CustomerModel customer = handleGuestCustomer(cartModel);
+
         //Update CartModel
         cartModel.setAdyenCseToken(adyenPaymentForm.getCseToken());
         cartModel.setAdyenDfValue(adyenPaymentForm.getDfValue());
+        cartModel.setUser(customer);
+        recalculateCart(cartModel);
```

Comment:
@maciek-marcinkowski I believe we can achieve the same result by omitting setUser (and also recalculateCart and CommerceCartService) since the CustomerModel is already persisted and the PK is not expected to change.
What do you think?